rfc/rfc-79/rfc-79.md

```diff
@@ -0,0 +1,314 @@
+goal (B) with respect to concurrent workers.
+
+The below visualization shows the flow for cancellable table service plans (steps that are already in existing table
+service flow are grey-ed out)
+
+![cancel table service lifecycle with lock (5)](https://github.com/user-attachments/assets/c05dae68-4330-4d85-b29e-e3e47754509e)
+
+Having this new .hoodie/.cancel folder (in addition to only having the .aborted state) is needed not only to allow any
+caller to forcebily block an instant from being committed, but also to prevent the need for table service workers to
+also perform write conflict detection (that ingestion already will perform) or unnecessarily re-attempt execution of the
+instant if it has been already been requested for cancellation but not succefully aborted yet. The below visualized
+scenario shows how this clustering attempt will "short circuit" in this manner by checking /.cancel to see if clustering
+execution should even proceed. This scenario also includes an example of concurrent writers to show how transaction and
+heartbeating in the above proposed flow will allow correct behavior even in the face of concurrent writers attempting to
+execute and/or cancel the instant.
+
+![cancel flow table serivce (1)](https://github.com/user-attachments/assets/f130f326-952f-49eb-bdbb-b0b34206f677)
+
+Aside from modifications to the clustering execution flow, a new pair of cancel APIs request_cancel and execute_abort
+will be added for all writers to use. They will allow a writer to request an instant to be cancelled (by adding it to
+/.cancel) and transition an instant to the terminal .aborted state, respectively.
+
+The new cancel API request_cancel will perform the following steps
+
+1. Start a transaction
+2. Reload active timeline
+3. If instant is already comitted, throw an exception
+4. If instant is already aborted, exit without throwing exception
+5. create a file under /.cancel with the target instant name
+6. End the transaction
+
+If this call succeeds, then the caller can assume that the target instant will not be commited and can only transitioned
+to .aborted state from that point onwards.
+
+The other API execute_abort will be added which allows a writer to explictly abort the target instant that has already
+had its cancellation requested (by adding the instant under /.cancel). Note that this API will also do cleaning up all
+leftover files in the process. Unlike request_cancel though, it cannot be completed if another writer is still executing
+the instant. In order to enforce this, heartbeating will be used to allow writers to infer whether another instant is
+being executed by a table service writer. The execute_abort API will perform the following steps
+
+1. Start a transaction
+2. Reload active timeline
+3. If instant has been already aborted, exit without throwing exception. Cleanup file from /.cancel if still present
```

Comment:
we might need to account for below scenario. 

our clustering table service can go through multiple attempts in general. Which means, there could be multiple rollbacks seen in timeline. so, when processing a request for cancellation, we need to deduce whether a rollback seen for a cancellable clustering plan happened before the cancel request or after. 

So that, we can decide whether to trigger another rollback or just proceed onto to moving the state to "abort"